src/components/QuranReader/index.tsx

```diff
@@ -114,11 +121,10 @@ const QuranReader = ({
   // if we are fetching the data (this will only happen when the user has changed the default translations/tafsirs so the initialData will be set to null).
   if (!data) {
     return (
-      <Loading
-        containerClassName={styles.container}
-        visibleSideBarClassName={styles.withVisibleSideBar}
-        isSideBarVisible={isSideBarVisible}
-      />
+      <>
+        <QuranReaderHeaderSkeleton />
+        <TranslationViewSkeleton />
```

Comment:
This should be based on the reading preference e.g. `ReadingViewSkeleton` when `isReadingPreference` is true